datafusion/sqllogictest/test_files/repartition.slt

```diff
@@ -121,9 +121,9 @@ Limit: skip=0, fetch=5
 --Filter: sink_table.c3 > Int16(0)
 ----TableScan: sink_table projection=[c1, c2, c3]
 physical_plan
-GlobalLimitExec: skip=0, fetch=5
---CoalescePartitionsExec
-----CoalesceBatchesExec: target_batch_size=8192
+CoalescePartitionsExec
+--GlobalLimitExec: skip=0, fetch=5
```

Comment:
I think this change is not correct. We cannot push down `GlobalLimitExec` though `CoalescePartitionsExec`.  However, we can convert following pattern
```
GlobalLimitExec: skip=0, fetch=5
--CoalescePartitionsExec
```
into
```
GlobalLimitExec: skip=0, fetch=5
--CoalescePartitionsExec
----LocalLimitExec: skip=0, fetch=5
```
if skip is larger than 0. `LocalLimitExec` should still have `skip=0`.

Reply:
Got it so the CoalescePartitionExec will also be a pushdown terminator and we just add a new LocalLimitExec below it.

Reply:
I don't really understand the reason to add an extra LocalLimitExec, I think we could simply add a global fetch when hit the pattern.

Reply:
I agree, that would be better.